configs/components/ruby-2.1.9.rb

```diff
@@ -5,20 +5,18 @@
   pkg.mirror "#{settings[:buildsources_url]}/ruby-#{pkg.get_version}.tar.gz"
 
   if platform.is_windows?
-    # elevate.exe is not used in the PDK
-    # pkg.add_source "#{settings[:buildsources_url]}/windows/elevate/elevate.exe", sum: "bd81807a5c13da32dd2a7157f66fa55d"
-    # pkg.add_source "file://resources/files/windows/elevate.exe.config", sum: "a5aecf3f7335fa1250a0f691d754d561"
     pkg.add_source "file://resources/files/ruby_219/windows_ruby_gem_wrapper.bat"
   end
 
-  # PDK: we don't need the additional metadata to replace other packages
-  # pkg.replaces 'pe-ruby'
-  # pkg.replaces 'pe-ruby-mysql'
-  # pkg.replaces 'pe-rubygems'
-  # pkg.replaces 'pe-libyaml'
-  # pkg.replaces 'pe-libldap'
-  # pkg.replaces 'pe-ruby-ldap'
-  # pkg.replaces 'pe-rubygem-gem2rpm'
+  if settings[:runtime_project] == 'agent'
+    pkg.replaces 'pe-ruby'
+    pkg.replaces 'pe-ruby-mysql'
+    pkg.replaces 'pe-rubygems'
+    pkg.replaces 'pe-libyaml'
+    pkg.replaces 'pe-libldap'
+    pkg.replaces 'pe-ruby-ldap'
+    pkg.replaces 'pe-rubygem-gem2rpm'
```

Comment:
I think these replaces should move into the agent-runtime component in puppet-agent, from thinking on it more I think that seems like the most straightforward way to manage these.